util/src/main/java/org/killbill/billing/util/discovery/DefaultEurekaKillbillService.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2014-2018 Groupon, Inc
+ * Copyright 2014-2018 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.util.discovery;
+
+
+import com.google.inject.Inject;
+import com.netflix.appinfo.ApplicationInfoManager;
+import com.netflix.appinfo.InstanceInfo;
+
+import org.killbill.billing.platform.api.KillbillService;
+import org.killbill.billing.platform.api.LifecycleHandlerType;
+import org.killbill.billing.platform.api.LifecycleHandlerType.LifecycleLevel;
+import org.killbill.billing.platform.config.DefaultKillbillConfigSource;
+import org.killbill.billing.tenant.api.TenantKV.TenantKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DefaultEurekaKillbillService implements EurekaKillbillService {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultEurekaKillbillService.class);
+
+    private ApplicationInfoManager applicationInfoManager;
+
+    public static final String EUREKA_SERVICE_NAME = "eureka-service";
+
+    @Inject
+    public DefaultEurekaKillbillService(ApplicationInfoManager applicationInfoManager) {
+        this.applicationInfoManager = applicationInfoManager;
+    }
+
+    @Override
+    public String getName() {
+        return EUREKA_SERVICE_NAME;
+    }
+
+    @LifecycleHandlerType(LifecycleLevel.INIT_SERVICE)
+    public synchronized void initialize() {
+        logger.info("--------------> Setting Eureka Status to UP <----------------");
+        applicationInfoManager.setInstanceStatus(InstanceInfo.InstanceStatus.UP);
```

Comment:
`INIT_SERVICE` is very early in the initialization sequence (https://github.com/killbill/killbill-platform/blob/7039977db0b9618a09454673b983c699bbc16740/platform-api/src/main/java/org/killbill/billing/platform/api/LifecycleHandlerType.java#L59), Kill Bill wouldn't be ready to accept requests yet.

Also, I don't think a new Kill Bill service is the way to go for this. Instead, this should happen at the end of the Guice initialization sequence, when we enable the healthcheck: https://github.com/killbill/killbill-platform/blob/7039977db0b9618a09454673b983c699bbc16740/server/src/main/java/org/killbill/billing/server/filters/KillbillGuiceFilter.java#L50-L52